src/core/utils/hooks/useContainerWidth.test.tsx

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+// import React from 'react';
+import { useContainerWidth } from './useContainerWidth';
+import * as UseResizeObserver from './useResizeObserver';
+import { renderHook, act } from '@testing-library/react-hooks';
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+afterEach(() => {
+  jest.restoreAllMocks();
+});
+
+it('should set initial size', () => {
+  const { result } = renderHook(() => useContainerWidth());
+  const element = document.createElement('div');
+  Object.defineProperties(element, {
+    getBoundingClientRect: { value: () => ({ width: 100 }) },
+  });
+  act(() => {
+    result.current[0](element);
+  });
+
+  expect(result.current[1]).toEqual(100);
+});
+
+it('should update value when resized', () => {
+  let triggerResize: (size: DOMRectReadOnly) => void = jest.fn();
+  jest
+    .spyOn(UseResizeObserver, 'useResizeObserver')
+    .mockImplementation((onResize) => {
+      triggerResize = onResize;
+      return [
+        jest.fn(),
+        ({ disconnect: jest.fn() } as unknown) as ResizeObserver,
+      ];
+    });
+  const { result } = renderHook(() => useContainerWidth());
+  const element = document.createElement('div');
```

Comment:
cant you have this element mocked before test and reuse it in every test case?

Reply:
Or you can just have an object with prop `getBoundingClientRect`.

Reply:
it's only used in two tests so I thought abstraction would not be that useful

Reply:
I count 3 uses, so..

Reply:
ok added helper